data-collection/src/main/java/refactoringml/util/RefactoringUtils.java

```diff
@@ -320,4 +319,40 @@ public static int refactoringTypeInNumber(Refactoring refactoring) {
 	public static boolean isStudied(Refactoring refactoring){
 		return refactoringTypeInNumber(refactoring) >= 0;
 	}
+
+	public static Set<ImmutablePair<String, String>> refactoredFilesAndClasses(Refactoring refactoring, Set<ImmutablePair<String, String>> classes) {
+		/**
+		 * if only one class is the origin of the refactoring (like most of them),
+		 * just return the current list
+ 		 */
+		if(classes.size() == 1)
+			return classes;
+
+		/**
+		 * if it's one of the refactorings below, this is a class-level refactoring
+		 * so let's return the list with all of classes, as we want data points for each
+		 * of these classes
+ 		 */
+		boolean extractSuperClassOrInterface = refactoring instanceof ExtractSuperclassRefactoring;
+		boolean moveSourceFolder = refactoring instanceof MoveSourceFolderRefactoring;
+		boolean renamePackageFolder = refactoring instanceof RenamePackageRefactoring;
+		if(extractSuperClassOrInterface || moveSourceFolder || renamePackageFolder)
+			return classes;
+
+		/**
+		 * if we get to here, we have in our hands a refactoring like
+		 * Move and Inline, or Extract and Move Method, which might have more than a single
+		 * class as origin for the refactoring.
+		 *
+		 * We, thus, would have to parse the refactoring description.
+		 * However, RMiner's implementation currently use a LinkedHashSet<>, which keeps
+		 * the order of insertion. In its implementation, the first element in the list
+		 * is always the origin. For now, we hope this won't change in RMiner.
+		 */
+		Iterator<ImmutablePair<String, String>> it = classes.iterator();
+		it.hasNext();
```

Comment:
Why do you call hasNext here?

Reply:
Oh, my, I was thought we needed to call `hasNext()` to set the pointer to the next element. I just tested here and you actually don't. You just changed my 15 years assumption!